private-templates-service/client/src/AuthenticatedComponent.tsx

```diff
@@ -0,0 +1,34 @@
+import React from "react";
+import { Redirect } from "react-router-dom";
+import { UserType } from "./App";
+import Welcome from "./components/Welcome/Welcome";
+
+interface State {
+}
+interface AuthProps {
+  isAuthenticated: boolean,
+  user?: UserType,
+  authButtonMethod: () => Promise<void>
+}
+
+export function requireAuthentication(Component: any) {
+
+  return class AuthenticatedComponent extends React.Component<AuthProps, State> {
```

Comment:
Since state is just an empty interface, you can just provide props - state isn't enforced